platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -256,62 +179,146 @@ private static KeyStore getKeyStore(String location, String password) {
+    }
+
+    return httpClientBuilder;
+  }
+
+  /**
+   * This method retrieves the system property using elevated privileges. The information returned
+   * by this method is visible to any object that can call the method. The supported cipher suites
+   * do not need this protection and therefore this method is not leaking sensitive information.
+   *
+   * @return supported cipher suites as an array
+   */
+  public static String[] getSupportedCipherSuites() {
+    return commaSeparatedToArray(HTTPS_CIPHER_SUITES);
+  }
+
+  /**
+   * This method retrieves the system property using elevated privileges. The information returned
+   * by this method is visible to any object that can call the method. The supported protocols do
+   * not need this protection and therefore this method is not leaking sensitive information.
+   *
+   * @return supported cipher suites as an array
+   */
+  public static String[] getSupportedProtocols() {
+    return commaSeparatedToArray(HTTPS_PROTOCOLS);
+  }
+
+  private boolean useBasicAuth() {
+    return Boolean.getBoolean("solr.useBasicAuth");
+  }
+
+  private String getCoreUrl(String coreName) {
+    return getSolrUrl() + "/" + coreName;
+  }
+
+  private static String getSolrUrl() {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL));
+  }
+
+  private CredentialsProvider getCredentialsProvider() {
+    CredentialsProvider provider = new BasicCredentialsProvider();
+    UsernamePasswordCredentials credentials =
+        new UsernamePasswordCredentials(
+            getProperty("solr.username"), getProperty("solr.credentials"));
+    provider.setCredentials(AuthScope.ANY, credentials);
+    return provider;
+  }
+
+  private static String getCoreDataDir(String coreName) {
+    return concatenatePaths(getCoreDir(coreName), "data");
+  }
+
+  private static String getCoreDir(String coreName) {
+    String solrDir = getProperty(SOLR_DATA_DIR);
+
+    if (StringUtils.isEmpty(solrDir)) {
+      throw new MissingResourceException(
+          "Cannot create Solr client. Missing data directory", "System", SOLR_DATA_DIR);
+    }
+
+    return concatenatePaths(solrDir, coreName);
+  }
+
+  private static String concatenatePaths(String first, String more) {
+    return Paths.get(first, more).toString();
+  }
+
+  private static String getProperty(String propertyName) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<String>) () -> System.getProperty(propertyName));
+  }
+
   private static boolean solrCoreExists(SolrClient client, String coreName)
       throws IOException, SolrServerException {
     CoreAdminResponse response = CoreAdminRequest.getStatus(coreName, client);
     return response.getCoreStatus(coreName).get("instanceDir") != null;
   }
+
+  private static String[] commaSeparatedToArray(String commaDelimitedString) {
+    // Consume whitespace
```

Comment:
The previous tokenizer would treat whitespace as part of the name of an element.  "C1,   C2" would become "C1"  and "  C2". (there are spaces before C2). I changed the regex to consume whitespace so that "C1,   C2" --> "C1", "C2" (the spaces before C2 are gone).
I left the comment in just in case someone was affected by the behavior change.
I am not sure if the comment is helpful or not. What is your opinion?